xbmc/utils/IBufferObject.h

```diff
@@ -15,12 +15,13 @@ class IBufferObject
 public:
   virtual ~IBufferObject() = default;
 
-  virtual bool CreateBufferObject(int width, int height) = 0;
-  virtual void DestroyBufferObject() { }
+  virtual bool CreateBufferObject(int format, int width, int height) = 0;
+  virtual void DestroyBufferObject() = 0;
   virtual uint8_t *GetMemory() = 0;
-  virtual void ReleaseMemory() { }
-  virtual int GetFd() { return -1; }
+  virtual void ReleaseMemory() = 0;
+  virtual int GetFd() = 0;
   virtual int GetStride() = 0;
+  virtual uint64_t GetModifier() = 0;
```

Comment:
To demonstrate, here's what the interface looks like: https://github.com/xbmc/xbmc/blob/6afe0b6edf38220b19e7f9d8a2bbec8be7eb7a2a/xbmc/utils/IBufferObject.h

some constructive criticism, I see ambiguity in the interface. `width` and `height` are obvious, but I'm not sure how an int represents a pixel format. `GetFd()` returns a file descriptor, which I'm familiar with from unix programming, but I'm not sure what API the file descriptor comes from, or what operations I'm allowed to perform on it. Now we're adding a modifier, but how does modification related to 64 bits?

Full doxy would make this interface rock.

Reply:
Full doxy added!